hibernate-reactive-core/src/main/java/org/hibernate/reactive/adaptor/impl/ResultSetAdaptor.java

```diff
@@ -727,9 +744,8 @@ public Blob getBlob(String columnLabel) {
 			return null;
 		}
 
-		byte[] bytes = value instanceof String
-				? ( (String) value ).getBytes()
-				: ( (Buffer) value ).getBytes();
+		byte[] bytes = value instanceof String ? ( (String) value ).getBytes()
+				: value instanceof byte[] ? (byte[]) value : ( (Buffer) value ).getBytes();
 		return BlobProxy.generateProxy( bytes );
```

Comment:
I would prefer to see this written out as a chain of `if`s with a specific case to handle the possibility that it's none of the types you're expecting.

A CCE from the typecast to `Buffer` isn't a good way to report the unexpected type.